### PR TITLE
chore: (2.34) Refactor sync implementation

### PR DESCRIPTION
**Pull request by huzvak**
- Clean the implementation of sync jobs
- Prepare for adding a paging feature for Completeness sync
- Prepare for adding new job parameteres for Metadata, Completeness and Data Values sync

DHIS2-6062

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/TrackerSynchronization.java [hunk 1]
```diff
@@ -65,6 +64,8 @@
     private final RestTemplate restTemplate;
     private final RenderService renderService;
 
+    private TrackedEntityInstanceQueryParams queryParams;
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/TrackerSynchronization.java]: We are putting status in a component, that could lead to some strange behaviour in concurrent calls.
Is there a way to create it inside the methods or passing it as a parameter?

Comment (huzvak) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/TrackerSynchronization.java]: The code is runned sequentially. It is not run concurrently and it is not built to be run concurrently. If there is a new requirement to run it concurrently, then the whole logic needs to be revised and probably significantly rewritten.

Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/TrackerSynchronization.java]: 👍

Comment (luciano-fiandesio) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/TrackerSynchronization.java]: @huzvak @enricocolasante I think that Enrico is correct in highlighting that variable. If 2 threads will use that bean, the variable will get overridden by the second call.

Comment (huzvak) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/sync/TrackerSynchronization.java]: Yes. I agree. But in Scheduler, there is a logic that there should never be 2 instances/threads of sync job running at the same time. But I will take a look into it anyway and will improve it in next commit and PR.

**Pull request merged**